pkg/asset/machines/openstack/machines.go

```diff
@@ -174,9 +174,13 @@ func generateProvider(clusterID string, platform *openstack.Platform, mpool *ope
 	return &spec, nil
 }
 
-func checkNetworkExtensionAvailability(cloud, alias string) (bool, error) {
-	opts := &clientconfig.ClientOpts{
-		Cloud: cloud,
+func checkNetworkExtensionAvailability(cloud, alias string, opts *clientconfig.ClientOpts) (bool, error) {
+	if opts == nil {
+		opts = &clientconfig.ClientOpts{
+			Cloud: cloud,
+		}
+	} else {
+		opts.Cloud = cloud
 	}
```

Comment:
```suggestion
	if opts == nil {
		opts = &clientconfig.ClientOpts{}
	}
	opts.Cloud = cloud
```
Although, I would personally rather see callers pass in a non-nil, zero `ClientOpts` rather than nil.